metabulo/normalization.py

```diff
@@ -30,9 +38,9 @@ def sum(table):
     return 1000 * table.div(table.sum(axis=1), axis=0)
 
 
-def reference_sample(table):
-    return table.iloc[0].sum() * table.div(table.sum(axis=1), axis=0)
+def reference_sample(table, argument):
+    return table.loc[argument].sum() * table.div(table.sum(axis=1), axis=0)
 
 
-def weight_volume(table):
-    return 100 * table.div(table.iloc[:, 0], axis=0)
+def weight_volume(table, argument, sample_metadata):
+    return 100 * table.div(sample_metadata.loc[:, argument], axis=0)
```

Comment:
This could throw or return NaN's if the metadata is non-numeric or zero with little to no indication to the user what has gone wrong.

Ideally, I think this would be a validation error on `PUT .../normalization`; however, that does lead to a possibility for the normalization argument to be valid before a column/row label change in the cleanup.  I'm inclined to ignore this problem for now because the changes to the data model I'm working on will probably reset the normalization/scaling/transforms completely.

Reply:
That being the case, I'm not super inclined to go down path of creating UI for handling these edge cases if it's going to behave differently in the near future.

Should I close this PR and wait on the model changes?

Reply:
I think it's okay to merge in and avoid worrying about edge cases.

Speculatively, I'm headed toward keeping the cleanup, imputation, column/row labeling where it is.  Then adding a new model for scaling, normalization, and transforms.  This new model would avoid all of the costly validation and row/column lookups that occur in the cleanup.

Right now, I'm not sure about how to generate the new model... Keeping the API exactly the same would require generating it on GET requests because the validation is done lazily.  I'm probably going to end up with some sort of explicit POST indicating clean up is done.  Whether we move the transformation endpoints to a new resource to make the change explicit on the API is still up in the air.

This is all just to say, the server side of this will just get moved directly to a new model.  The REST API will look mostly the same.  I don't see a reason to make validation lazy post-cleanup stage.  Errors should be thrown in the mutation endpoint.